src/main/java/uk/gov/pay/api/model/SelfAndNextLinks.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.pay.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.List;
+
+import static javax.ws.rs.HttpMethod.GET;
+import static uk.gov.pay.api.model.SelfLinks.SELF;
+
+@ApiModel(value = "selfAndNextLinks", description = "Resource self and next links of a Payment")
+public class SelfAndNextLinks {
```

Comment:
This is a case where we could make use of inheritance and then simply annotate the methods required for documentation.
This will move the documentation annotation into the concrete classes, but I think it is a lesser evil than having all this boiler plate code for documentation sake.